### PR TITLE
Adds package gdal and 16 other dependency packages

### PR DESCRIPTION
**Pull request by arc9693**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

What does the PR accomplish, why was it needed?
Adds the following packages in SPECS-EXTENDED:
1. gdal
2. freexl
3. libgeotiff
4. proj
5. gpsbabel
6. hdf
7. netcdf
8. libkml
9. uriparser
10. blosc
11. qt5-qtserialport
12. shapelib
13. CharLS
14. cfitsio
15. ogdi
16. libgta
17.  liblerc

###### Does this affect the toolchain?  

**NO**

###### Test Methodology

- Pipeline build id: https://dev.azure.com/mariner-org/mariner/_build/results?buildId=410746&view=results,
- https://dev.azure.com/mariner-org/mariner/_build/results?buildId=410771&view=results

### CODE REVIEW

File: SPECS-EXTENDED/xerces-c/xerces-c.spec [hunk 1]
```diff
@@ -0,0 +1,260 @@
+Summary:        Validating XML Parser
+Name:           xerces-c
```
Comment (mfrw) [inline SPECS-EXTENDED/xerces-c/xerces-c.spec:2]: We already have this:
- 
https://github.com/microsoft/CBL-Mariner/pull/5976

File: SPECS-EXTENDED/hdf/hdf.spec [hunk 2]
```diff
@@ -0,0 +1,578 @@
+# No more Java on i686
+%ifarch %{java_arches}
+%bcond_without java
+%else
+%bcond_with java
+%endif
+Summary:        A general purpose library and file format for storing scientific data
+Name:           hdf
+Version:        4.2.15
```
Comment (mfrw) [inline SPECS-EXTENDED/hdf/hdf.spec:9]: I think we have `hdf5`, does not suffice for our usecase ?
Reference: https://github.com/microsoft/CBL-Mariner/blob/2.0/SPECS/hdf5/hdf5.spec#L14

Comment (arc9693) [reply SPECS-EXTENDED/hdf/hdf.spec:9]: No @mfrw  
Reference : https://gdal.org/development/building_from_source.html#hdf4

File: SPECS-EXTENDED/xerces-c/xerces-c.spec (hunk 1 above)
Comment (arc9693) [reply SPECS-EXTENDED/xerces-c/xerces-c.spec:2]: I know, it was not merged back then. I will rebase and remove this.

File: cgmanifest.json [hunk 3]
```diff
@@ -29763,7 +29923,7 @@
         "other": {
           "name": "xerces-c",
           "version": "3.2.4",
-          "downloadUrl": "https://downloads.apache.org/xerces/c/3/sources/xerces-c-3.2.4.tar.xz"
+          "downloadUrl": "https://downloads.apache.org/xerces/c/3/sources/xerces-c-3.2.4.tar.gz"
```
Comment (mfrw) [inline cgmanifest.json:29926]: I think it should be `xz` as per the [spec](https://github.com/microsoft/CBL-Mariner/blob/main/SPECS/xerces-c/xerces-c.spec#L9)

**Review by mfrw — CHANGES_REQUESTED** [review]
Can we break this PR in smaller chunks.
This feels a little difficult to review.
Also, think `revert` case.
If one of the packages needs to be reverted for some reason, reverting this `squashed-merge` would revert the entire 17-18 packages.

**Review by kanikanema — COMMENTED** [review]
Thanks for the PR. To work out all the dependencies is a lot of tedious work.
A review of this size is very very hard to review properly. Could you split it into smaller chunks say, the leaf node specs, then the next set of dependencies and so on till you reach gdal?
Though. a little bit of extra work git does make it easy to do something like this.

**Comment by arc9693** [comment]
> Can we break this PR in smaller chunks. This feels a little difficult to review. Also, think `revert` case. If one of the packages needs to be reverted for some reason, reverting this `squashed-merge` would revert the entire 17-18 packages.

I started out by creating singled out PRs, was advised to create single one with all dependencies.
I agree with the revert case. Sure, I will break it down for easy review.

**Comment by mfrw** [comment]
> > Can we break this PR in smaller chunks. This feels a little difficult to review. Also, think `revert` case. If one of the packages needs to be reverted for some reason, reverting this `squashed-merge` would revert the entire 17-18 packages.
> 
> I started out by creating singled out PRs, was advised to create single one with all dependencies. I agree with the revert case. Sure, I will break it down for easy review.

I would say .. group all packages that depend on each other in a single PR .. and maybe the leaf nodes can go as a seperate PR.

**Pull request closed without merging**